research/error-correction/hamming.py

```diff
@@ -0,0 +1,75 @@
+from copy import copy
+from math import ceil, log2
+# A very basic implementation of the Hamming code function in Python
+
+# https://stackoverflow.com/a/57025941
+def power_of_two(n):
```

Comment:
Confusing name
Rename to `is_power_of_two`?

Reply:
Agreed, changed